frontend/public/components/start-guide.tsx

```diff
@@ -10,50 +11,38 @@ import { ProjectModel, RoleModel, StorageClassModel } from '../models';
 
 const WHITELIST = [RoleModel.kind, StorageClassModel.kind];
 
-const seenGuide: string = 'seenGuide';
-
-export class StartGuide extends React.Component<StartGuideProps, StartGuideState> {
-  constructor(props) {
-    super(props);
-
-    // TODO: The dismissable logic is not currently being used, but we'll
-    // probably want a general start guide for OpenShift separate from the no
-    // projects message. Leaving this for now.
-    this.dismiss = this.dismiss.bind(this);
-    let {visible} = props;
-    try {
-      visible = visible || !localStorage.getItem(seenGuide);
-    } catch (ignored) {
-      // ignored
-    }
-    this.state = {
-      visible,
-    };
+export const StartGuide: React.FC<StartGuideProps> = (props) => {
+  let {dismissKey, startGuide, visible} = props;
+  try {
+    visible = visible || !localStorage.getItem(dismissKey);
```

Comment:
I don't think we need the visible prop? It should always be true. But we should skip the `localStorage` check if there is no dismiss key.

```suggestion
    visible = !dismissKey || !localStorage.getItem(dismissKey);
```